src/e2e/java/teammates/e2e/cases/e2e/InstructorCourseDetailsPageE2ETest.java

```diff
@@ -0,0 +1,126 @@
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/InstructorCourseDetailsPageE2ETest.json");
+        student = testData.students.get("charlie.tmms@ICDetailsE2eT.CS2104");
+        if (!TestProperties.isDevServer()) {
+            student.email = TestProperties.TEST_STUDENT1_ACCOUNT;
+        }
+
+        removeAndRestoreDataBundle(testData);
+        course = testData.courses.get("ICDetailsE2eT.CS2104");
+        fileName = "/" + course.getId() + "_studentList.csv";
+    }
+
+    @BeforeClass
+    public void classSetup() {
+        deleteDownloadsFile(fileName);
+    }
+
+    @Test
+    public void allTests() {
+        AppUrl detailsPageUrl = createUrl(Const.WebPageURIs.INSTRUCTOR_COURSE_DETAILS_PAGE)
+                .withUserId(testData.instructors.get("ICDetailsE2eT.instr").googleId)
+                .withCourseId(course.getId());
+        InstructorCourseDetailsPage detailsPage = loginAdminToPage(detailsPageUrl, InstructorCourseDetailsPage.class);
+
+        ______TS("verify loaded details");
+        InstructorAttributes[] instructors = {
+                testData.instructors.get("ICDetailsE2eT.instr"),
+                testData.instructors.get("ICDetailsE2eT.instr2"),
+        };
+        StudentAttributes[] students = {
+                testData.students.get("alice.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("charlie.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("benny.tmms@ICDetailsE2eT.CS2104"),
+                testData.students.get("danny.tmms@ICDetailsE2eT.CS2104"),
+        };
+
+        verifyCourseDetails(detailsPage, course, instructors, students);
+        detailsPage.verifyNumStudents(4);
+        detailsPage.verifyStudentDetails(students);
+
+        ______TS("send invite");
+        detailsPage.sendInvite(student);
+
+        detailsPage.verifyStatusMessage("An email has been sent to " + student.getEmail());
+        String expectedEmailSubject = "TEAMMATES: Invitation to join course ["
+                + course.getName() + "][" + course.getId() + "]";
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
+
+        ______TS("remind all students to join");
+        detailsPage.remindAllToJoin();
+
+        detailsPage.verifyStatusMessage("Emails have been sent to unregistered students.");
+        verifyEmailSent(student.getEmail(), expectedEmailSubject);
+
+        ______TS("download student list");
+        detailsPage.downloadStudentList();
+        String status = student.googleId.isEmpty() ? "Yet to Join" : "Joined";
+        String lastName = student.getName().split(" ")[1];
+        String[] studentInfo = { student.getTeam(), student.getName(), lastName, status, student.getEmail() };
+        List<String> expectedContent = Arrays.asList("Course ID," + course.getId(),
+                "Course Name," + course.getName(), String.join(",", studentInfo));
+        verifyDownloadedFile(fileName, expectedContent);
```

Comment:
I find it odd that we only check that one student is in the downloaded file. Shouldn't we check and ensure all the student's are there?

`verifyDownloadedFile` name is a little misleading - I think a better name is `verifyDownloadedFileContainsContent` (since all it does is check if something is inside - it is not an equality check)

Reply:
This is my suggestion. If you want to ensure all students are there, that's the job of a unit test. Here is just a list of some representative content, although admittedly more wouldn't hurt.

Reply:
Yes agreed, a unit test is better for the above check